code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -227,6 +227,15 @@ var/global/list/simplemob_icon_bitflag_cache = list()
 	do_delayed_life_action()
 	performing_delayed_life_action = FALSE
 
+/mob/living/simple_animal/proc/turf_is_safe(turf/target)
+	if(!istype(target))
+		return FALSE
+	if(target.is_open() && target.has_gravity() && !can_overcome_gravity())
+		return FALSE
+	if(is_aquatic != !target.submerged())
```

Comment:
The double negative here makes me worry this check is inverted. This means an aquatic creature would consider a dry tile safe and a wet tile dry.